knowledge_repo/app/models.py

```diff
@@ -458,6 +460,19 @@ def update_metadata_from_kp(self, kp):
 
         self.status = kp.status
 
+        if headers.get('users_allowed_to_view', ''):
+            self.special_permissions = 1
+            # comma separated list, add it into the Permission table
+            users_allowed_to_view = headers.get('users_allowed_to_view')
+            for user in users_allowed_to_view.split(","):
+                user = (db_session.query(User).filter(User.username == user.strip()).first())
+                if not user:
+                    user = User(username=user)
+                    db_session.add(user)
+                perm = Permission(user_id=user.id, post_uuid=self.uuid)
+                db_session.add(perm)
+                db_session.commit()
```

Comment:
It looks like there's not a way to delete permissions. Can we make it so if the header isn't there, any permissions are deleted?